### PR TITLE
add clang-uml for making class diagrams

### PR DESCRIPTION
**Pull request by davidlange6**
Has proven useful for building class diagrams from cmssw
https://clang-uml.github.io

pretty lightweight to keep around. Depends on yaml-cpp which we already have. I'm using close to the tip of the master branch as it contains a bug fix for discovering libraries (eg, yaml-cpp) in non-standard locations (eg, the cmssw use case)

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc12

**Comment by smuzaffar** [comment]
please test for el8_ppc64le_gcc12

**Comment by davidlange6** [comment]
@smuzaffar - is this warning a common issue on aarch64 builds? Its not totally obvious how to properly fix it

**Comment by smuzaffar** [comment]
> @smuzaffar - is this warning a common issue on aarch64 builds? Its not totally obvious how to properly fix it

Not really, may be some part of code is conditionally compiled for aarch64. I would suggest to add
```
%ifarch aarch64
-DCMAKE_CXX_FLAGS="-Wno-sign-compare" \
%endif
```

File: clang-uml.spec [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+CMAKE_PREFIX_PATH=${YAML_CPP_ROOT}/lib64/cmake/yaml-cpp/ cmake ../%{n}-%{realversion} \
+  -G Ninja \
+  -DCMAKE_INSTALL_PREFIX:PATH="%i" \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_CXX_FLAGS="" \
+  -DCMAKE_EXE_LINKER_FLAGS="" \
+  -DLLVM_VERSION= \
+  -DCMAKE_VERBOSE_MAKEFILE=true \
+  -DGIT_VERSION="0.4.1-9-g205d6de" \
+  -DLLVM_CONFIG_PATH=
```
Comment (smuzaffar) [inline clang-uml.spec]: why empty values for `CMAKE_EXE_LINKER_FLAGS, LLVM_VERSION,LLVM_CONFIG_PATH` ? Are these really needed for can we remove these? You can add dependency on `llvm` to get llvm version and path (if needed)

Comment (davidlange6) [reply clang-uml.spec]: hum, I was copying
https://github.com/bkryza/clang-uml/blob/f3a0d98ee6fd0fffb4fa2a255f0c337f6d81db9d/Makefile#L33

I can try to add versions to make it more clear

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
please test for el8_aarch64_gcc12

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Comment by smuzaffar** [comment]
ignore tests-rejected with external-failure

this unit tests some times fails due to DAS glitch OR local proxy file being overridden
```
ERROR failed to parse X509 proxy: crypto/tls: failed to parse key: asn1: syntax error: sequence truncated
```

**Pull request merged**